app/lib/ability.rb

```diff
@@ -189,6 +188,9 @@ def initialize(user)
     end
 
     if user.site_coordinator?
+      can :manage, User, id: User.where(role: SiteCoordinatorRole.assignable_to_sites(user.role.sites)).pluck(:id)
+      can :manage, User, id: User.where(role: TeamMemberRole.assignable_to_sites(user.role.sites)).pluck(:id)
```

Comment:
I guess instead of including all those custom actions you could also just add a cannot after the lines like for example
```suggestion
      can :manage, User, id: User.where(role: TeamMemberRole.assignable_to_sites(user.role.sites)).pluck(:id)
      cannot :destroy, User
```